### ISSUE TITLE
Data Loss/Corruption - librarynotfoundexception

### ISSUE THREAD

**Issue by luisbaia** [manahl/arctic#741]
#### Arctic Version

```
# 1.74.0
```

#### Arctic Store

```
# VersionStore
```

#### Platform and version

Mongo version v3.6.3 deployed on a single aws instance

#### Description of problem and/or code sample that reproduces the issue

I have been using arctic for the last month. I have been uploading data to it every second or less. For two times, out of nowhere when reading data from it, I get a "librarynotfoundexception" from the data.  More specifically:

arctic.exceptions.LibraryNotFoundException: Library my_library was not correctly initialized in <Arctic at 0x7f06ef5b9e80, connected to MongoClient(host=[my_host], document_class=dict, tz_aware=False, connect=True, maxpoolsize=4, sockettimeoutms=600000, connecttimeoutms=2000, serverselectiontimeoutms=30000)>.

The first time, I tried to initialize the library again using:

store = Arctic(MONGO_HOST, app_name='arctic')
store.initialize_library(library=library)

Ultimately this led to a full data loss. 

Now, the same exception happened again (after almost two weeks of every second appends). Any ideas?

**Comment by luisbaia** [comment]
When I run:

store.list_libraries()

The libraries I had no longer appear.

**Comment by shashank88** [comment]
Hi, Do you have a sharded collection or did you rename the collection? Next time this happens can you go inside the mongo cluster and check the db and the collection... Sounds like a Mongo issue most likely

**Comment by luisbaia** [comment]
Every time I initialize a new library I get the warning:

"Library created, but couldn't enable sharding: no such command: 'enablesharding', bad cmd: '{ enablesharding: "arctic", lsid: { id: UUID("09b14d8f-30e7-4785-8842-ec704ce9f6bd") }, $db: "admin", $readPreference: { mode: "primaryPreferred" } }'. This is OK if you're not 'admin'"

Could this be related somehow?

**Comment by jamesblackburn** [comment]
That just indicates that you’re not running against a shared cluster (or don’t have sufficient privileges to enable sharping). 

Initialize_library shouldn’t be a destructive process even if the library already exists. If the library is going missing then I think something is calling delete library in arctic or otherwise dropping the mongo database. 

Do you have the mongo logs from around the time period of the issue? This should contain database and collection dropping events.

**Comment by luisbaia** [comment]
In the meantime I have created a new mongodb using the full AWS capabilities/services, rather than starting a single instance and manually installing mongo there and leave it running with the default configurations. Other services were running on that instance and I did not even pay attention if the folder mongo was storing data was not some directory whose data persistence was not guaranteed.

I will assume the data loss was due to one of the abose reasons. The last reason is the only one I can give that can explain this weird behaviour.

I will close this issue and assume that was the reason. Nevertheless thanks for the quick response.
